platforms/core-execution/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerDaemonLifecycleTest.groovy

```diff
@@ -169,7 +169,7 @@ class WorkerDaemonLifecycleTest extends AbstractDaemonWorkerExecutorIntegrationS
         succeeds "compileJava", "runInWorker"
 
         then:
-        sinceSnapshot().count("Started Gradle worker daemon") == 2
```

Comment:
I've checked, and the in-process spec is executed here; hence the one-less gradle worker

Reply:
❌ This breaks what we're checking here.  This is supposed to be checking that when there are two workers (WorkerTask and java compilation), we stop the WorkerTask worker but not the java compilation one.

Reply:
Okay, that means that there are two ways to do this:
 - A: By doing this change, as the java compiler becomes in-process
 - B: Forcing the java compiler to fork, so it will make a worker

I assume from your response that you would like to do option B.

Reply:
correct

Reply:
Now I noticed as you mentioned that the test explicitly asks for a worker, so something is probably wrong.

Reply:
Okay, eventually nothing was "wrong" in the current HEAD.
The fix in the spec selection, which corrected the erroneous implementation in the `isForking` branch solved this automatically.

Dropped the commits doing work on this.